script.go

```diff
@@ -397,6 +408,47 @@ func (p *Pipe) Exec(cmdLine string) *Pipe {
 	})
 }
 
+// Exec runs cmdLine as an external command, sending it the contents of the
+// pipe as input, and produces the command's standard output (see below for
+// error output). The effect of this is to filter the contents of the pipe
+// through the external command. It also allows passing context.Context making
+// it cancellable
+//
+// # Error handling
+//
+// If the command had a non-zero exit status, the pipe's error status will also
+// be set to the string “exit status X”, where X is the integer exit status.
+// Even in the event of a non-zero exit status, the command's output will still
+// be available in the pipe. This is often helpful for debugging. However,
+// because [Pipe.String] is a no-op if the pipe's error status is set, if you
+// want output you will need to reset the error status before calling
+// [Pipe.String].
+//
+// If the command writes to its standard error stream, this will also go to the
+// pipe, along with its standard output. However, the standard error text can
+// instead be redirected to a supplied writer, using [Pipe.WithStderr].
+func (p *Pipe) ExecContext(ctx context.Context, cmdLine string) *Pipe {
```

Comment:
I'm a bit worried by the code that was copy-pasted here. The code is good, but it would require to maintain two methods

I would have expected Exec to be refactored to call ExecContext with context.Background()